internal/services/network/network_security_group_resource.go

```diff
@@ -447,7 +455,7 @@ func flattenNetworkSecurityRules(rules *[]network.SecurityRule) []map[string]int
 					sgRule["source_port_ranges"] = set.FromStringSlice(*props.SourcePortRanges)
 				}
 
-				sgRule["protocol"] = string(props.Protocol)
+				sgRule["protocol"] = string(protocolMap[strings.ToLower(string(props.Protocol))])
```

Comment:
just an FYI - there's a Pandora PR to normalize this by default we'll be rolling out in the future too: https://github.com/hashicorp/pandora/pull/758